File: benchmark/output/lua-love/gemini/particle_burst.lua
local particle_xs = {}
local particle_ys = {}
local particle_vxs = {}
local particle_vys = {}
local particle_lifes = {}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Particle Burst")
end

function love.mousepressed(mx, my, button)
    if button == 1 then
        for i = 1, 10 do
            local angle = love.math.random() * 6.283
            local speed = love.math.random(50, 200)
            
            table.insert(particle_xs, mx)
            table.insert(particle_ys, my)
            table.insert(particle_vxs, math.cos(angle) * speed)
            table.insert(particle_vys, math.sin(angle) * speed)
            table.insert(particle_lifes, 2.0)
        end
    end
end

function love.update(dt)
    for i = #particle_lifes, 1, -1 do
        particle_lifes[i] = particle_lifes[i] - dt
        
        if particle_lifes[i] <= 0 then
            table.remove(particle_xs, i)
            table.remove(particle_ys, i)
            table.remove(particle_vxs, i)
            table.remove(particle_vys, i)
            table.remove(particle_lifes, i)
        else
            particle_xs[i] = particle_xs[i] + particle_vxs[i] * dt
            particle_ys[i] = particle_ys[i] + particle_vys[i] * dt
        end
    end
end

function love.draw()
    for i = 1, #particle_lifes do
        if particle_lifes[i] > 0 then
            love.graphics.rectangle("fill", particle_xs[i], particle_ys[i], 4, 4)
        end
    end
end

File: vibe-lang/benchmark/output/python-pygame/openai/particle_burst.py
import pygame
import sys
import random
import math

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Particle Burst")
clock = pygame.time.Clock()

particle_xs = []
particle_ys = []
particle_vxs = []
particle_vys = []
particle_lifes = []

def rand_float(min_val, max_val):
    return random.uniform(min_val, max_val)

def mousepressed(mx, my, button):
    for _ in range(10):
        angle = rand_float(0.0, 6.283)  # 0 to 2*pi
        speed = rand_float(50.0, 200.0)
        vx = math.cos(angle) * speed
        vy = math.sin(angle) * speed
        particle_xs.append(mx)
        particle_ys.append(my)
        particle_vxs.append(vx)
        particle_vys.append(vy)
        particle_lifes.append(2.0)

def update(dt):
    for i in range(len(particle_xs) - 1, -1, -1):
        particle_xs[i] += particle_vxs[i] * dt
        particle_ys[i] += particle_vys[i] * dt
        particle_lifes[i] -= dt
        if particle_lifes[i] <= 0.0:
            del particle_xs[i]
            del particle_ys[i]
            del particle_vxs[i]
            del particle_vys[i]
            del particle_lifes[i]

def draw():
    for i in range(len(particle_xs)):
        if particle_lifes[i] > 0.0:
            pygame.draw.rect(screen, (255, 255, 255), (int(particle_xs[i]), int(particle_ys[i]), 4, 4))

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mx, my = pygame.mouse.get_pos()
            mousepressed(mx, my, event.button)

    update(dt)

    screen.fill((0, 0, 0))
    draw()
    pygame.display.flip()

pygame.quit()
sys.exit()